projects/table-builder/src/lib/directives/multi-sort.directive.ts

```diff
@@ -13,56 +14,33 @@ import { scan, filter, map } from 'rxjs/operators';
 })
 export class MultiSortDirective extends MatSort implements OnInit, OnDestroy {
   @Output() readonly multiSortChange: EventEmitter<Sort[]> = new EventEmitter<Sort[]>();
-
-  @Input() rules$: Observable<Sort[]> = new Subject();
-
-  @Input() set  reSort (val){
-    this.initRules();
-  } 
   rules: Sort[] = [];
-  preSortRules: Sort[] = [];
   private SubRef: Subscription;
 
+  constructor(private state: TableStateManager) {
+    super();
+  }
+
   ngOnInit() {
-    this.SubRef = this.rules$.pipe(
-      scan((acc,curr)=>({prev: acc.curr, curr }),{prev: null, curr: null} as {prev: Sort[], curr: Sort[]}),
-      filter(acc => !acc.prev || acc.curr.some(sort => !acc.prev.some(prevSort => sort.active === prevSort.active && sort.direction === prevSort.direction))),
```

Comment:
I think I know the point of this filter. For some reason the rules observable was refiring multiple times, so we filter to check if it is a new set of rules. Does your refactor support creating a new set of rules? I am not sure if in the old way it was actually possible to pass in new rules

Reply:
It only fires if the store fires. So it should not be an issue. distinctUntilChanged() should handle that.